simplecpp.cpp

```diff
@@ -327,6 +327,15 @@ static void ungetChar(std::istream &istr, unsigned int bom)
         istr.unget();
 }
 
+static unsigned char prevChar(std::istream &istr, unsigned int bom)
+{
+    ungetChar(istr, bom);
+    ungetChar(istr, bom);
```

Comment:
This line introduces the following problem at least on Windows/MSVC 2017:
If a string constant starts exactly at the _INTERNAL_BUFSIZ = 4096 bytes (constant in ucrt\inc\corecrt_internal_lowio.h), simplecpp outputs
syntax error: No pair for character ("). Can't process file. File is either invalid or unicode, which is currently not supported.

Analysis: The problem occurs in new method prevChar that tries to unget 2 characters. When trying to unget the second character, the internal buffer is empty, so the call fails and sets badbit. The following call to readChar then also fails and sets failbit.

Example input file (GitHub does not support .cpp files, so please rename .txt to .cpp)
[StringConstant.txt](https://github.com/danmar/simplecpp/files/3017659/StringConstant.txt)

Reply:
Yeah, that function did not give my a warm and fuzzy feeling when I implemented it...

Can you test #160? I was not able to reproduce this on Linux. Also, this means that reading of utf-16 files is probably broken as well, since there is no guarantee that unget can be called twice in a row?